torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -1578,6 +1578,42 @@ Tensor binary_cross_entropy_target_backward(
   return grad_target;
 }
 
+Tensor binary_cross_entropy_double_backward_target(
+  const Tensor& grad,
+  const Tensor& grad_output,
+  const Tensor& self,
+  const Tensor& target,
+  const c10::optional<Tensor>& weight,
+  int64_t reduction
+) {
+  auto res = -grad * grad_output;
+
+  if (isDefined(weight)) {
+    if (!isTensorSubclassLike(weight.value())) {
+      res.mul_(weight.value());
+    } else {
+      res = res * weight.value();
+    }
+  }
+
+  auto denom = (1 - self).mul_(self);
+  {
+    at::NoGradGuard guard;
```

Comment:
Compliance wrt to other kernels, which clamp at `eps` in an autograd-agnostic way.

Reply:
Oh gotcha, also from discussion with @albanD offline, it seems like it is useful for avoiding setting gradients to zero for values that are clamped. Not sure the clamp is entirely for division by zero in this case though because negative values would also be clamped to eps?

Reply:
There should be no negative values, inputs for binary cross entropy are values between 0 and 1.